Search/bfsnew.cpp

```diff
@@ -0,0 +1,58 @@
+#include <iostream>
+using namespace std;
+int queue[5];
+int a[5];
+int root;
+int L;
+int R;
+int front=0;
+int end=0;
+void BFS();
+void push();
+void pop();
+
+int main()
+{
+	for(int i=0;i<5;i++)
+	{
+		cin>>a[i];
+	}
+	cout<<"\nPrinting in bfs order";
+	BFS();
+	
+	return 0;
+ } 
+ void BFS()
+ {
+
+ 	queue[0]=a[root];
+ 	end++;
+ 	while(front<5)
+ 	{
+ 		pop();
+ 		push();
```

Comment:
in BFS we don't just pop / push values into the queue.
We pop _(from front)_ the nodes / vertices that have been visited.
& push the nodes / vertices that are adjacent to the node in the front of the queue.